modules/dnn/test/test_layers.cpp

```diff
@@ -1583,6 +1583,134 @@ TEST(Layer_Test_Convolution, relu_fusion)
+
+        lp.blobs.push_back(conv_mat_1);
+        id_conv[0] = net.addLayer(lp.name, lp.type, lp);
+        net.connect(0, 0, id_conv[0], 0);
+    }
+
+    {
+        LayerParams lp;
+        lp.set("kernel_size", 3);
+        lp.set("num_output", 3);
+        lp.set("bias_term", false);
+        lp.type = "Convolution";
+        lp.name = "testConv_2";
+
+        lp.blobs.push_back(conv_mat_2);
+        id_conv[1] = net.addLayer(lp.name, lp.type, lp);
+        net.connect(0, 0, id_conv[1], 0);
+    }
+
+    int eltwise_id;
+    {
+        LayerParams lp;
+        lp.type = "Eltwise";
+        lp.name = "testLayer";
+        lp.set("operation", "sum");
+
+        eltwise_id = net.addLayer(lp.name, lp.type, lp);
+        net.connect(id_conv[0], 0, eltwise_id, 0);
+        net.connect(id_conv[1], 0, eltwise_id, 1);
+    }
+
+    {
+        LayerParams lp;
+        lp.set("bias_term", false);
+        lp.type = "Scale";
+        lp.name = "testScale_1";
+
+        lp.blobs.push_back(scale_mat_1);
+        net.addLayerToPrev(lp.name, lp.type, lp);
+    }
+
+    {
+        LayerParams lp;
+        lp.set("bias_term", false);
+        lp.type = "Scale";
+        lp.name = "testScale_2";
+
+        lp.blobs.push_back(scale_mat_2);
+        net.addLayerToPrev(lp.name, lp.type, lp);
+    }
+
+    {
+        LayerParams lp;
+        lp.type = "ReLU";
+        lp.name = "testReLU";
+        net.addLayerToPrev(lp.name, lp.type, lp);
+    }
+
+    int sz[] = {1, 1, 12, 12};
+    Mat input(4, &sz[0], CV_32F);
+    rng.fill(input, RNG::UNIFORM, Scalar(0), Scalar(1));
+
+    Backend backendId = get<0>(GetParam());
+    Target  targetId  = get<1>(GetParam());
+
+    net.setInput(input);
+    net.setPreferableBackend(backendId);
+    net.setPreferableTarget(targetId);
+    Mat output = net.forward();
+    Mat output_2d(output.size[2], output.size[3], CV_32F, output.ptr<float>());
+
+    const double l1 = (targetId == DNN_TARGET_OPENCL_FP16 || targetId == DNN_TARGET_MYRIAD) ? 3.1e-4 : 1e-5;
+    const double lInf = (targetId == DNN_TARGET_OPENCL_FP16 || targetId == DNN_TARGET_MYRIAD) ? 9.1e-4 : 1e-4;
+
+    Mat conv_out_1, conv_out_2;
+    cvtest::filter2D(
+        Mat(input.size[2], input.size[3], CV_32F, input.ptr<float>()),
+        conv_out_1, CV_32F,
+        Mat(conv_mat_1.size[2], conv_mat_1.size[3], CV_32F, conv_mat_1.ptr<float>()),
+        Point(-1,-1), 0, BORDER_DEFAULT
+    );
+    cvtest::filter2D(
+        Mat(input.size[2], input.size[3], CV_32F, input.ptr<float>()),
+        conv_out_2, CV_32F,
+        Mat(conv_mat_2.size[2], conv_mat_2.size[3], CV_32F, conv_mat_2.ptr<float>()),
+        Point(-1,-1), 0, BORDER_DEFAULT
+    );
+
+    const int pad = 1;
+    Mat result(
+        conv_out_1(cvtest::Rect(pad, pad, output_2d.size[0], output_2d.size[1])) +
+        conv_out_2(cvtest::Rect(pad, pad, output_2d.size[0], output_2d.size[1]))
+    );
+
+    result = result * scale_mat_1.at<float>(0, 0);
+    result = result * scale_mat_2.at<float>(0, 0);
```

Comment:
Maybe it's easier to create two networks but for one of them set `enableFusion(false)`?

Reply:
@dkurt changed.